jetty-client/src/main/java/org/eclipse/jetty/client/HttpRedirector.java

```diff
@@ -157,6 +159,14 @@ public Request redirect(Request request, Response response, Response.CompleteLis
             URI newURI = extractRedirectURI(response);
             if (newURI != null)
             {
+                boolean absolute = newURI.isAbsolute();
+                URI uri = request.getURI();
+                if (absolute && newURI.equals(uri) ||
+                        !absolute && Objects.equals(newURI.getPath(), uri.getPath()))
+                {
+                    fail(request, response, new HttpResponseException("Redirect to same URI: " + location, response));
```

Comment:
I don't think that an A->A redirect is obviously infinite!   It may be redirecting from http to https, or POST to GET, or changing some other state in the server that needs a retry.  I would count A->A exactly the same as A->B

Reply:
@gregw good point. I have updated the code, please re-review.